### ISSUE TITLE
Support Token Provider integration

### ISSUE THREAD

**Issue by danielmarbach** [Particular/NServiceBus.Transport.AzureServiceBus#194]
https://azuresdkdocs.blob.core.windows.net/$web/dotnet/Azure.Identity/1.4.0-beta.1/api/index.html#client-libraries-supporting-authentication-with-azure-identity

Once the transport is upgraded to the new SDK we should make it possible to support Token Provider integration

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/README.md#authenticating-with-azureidentity

and the more full-blown sample in Event Hubs

https://github.com/Azure/azure-sdk-for-net/tree/Azure.Identity_1.4.0-beta.1/sdk/eventhub/Azure.Messaging.EventHubs#using-an-active-directory-principal-with-the-event-hub-clients

**Comment by alhardy** [comment]
Sas about to log an issue to the same effect, this would be a great addition

**Comment by yvesgoeleven** [comment]
This capability is only available on the future servicebus sdk, https://docs.microsoft.com/en-us/dotnet/api/overview/azure/messaging.servicebus-readme-pre#authenticating-with-azureidentity

We will need to wait until it has sufficient feature parity before we can start leveraging it.

**Comment by jbogard** [comment]
*bump* it's available now, you can authenticate against Azure.Identity. Was doing a checklist of all our dependencies and this was the last one needing updating away from AppAuthentication

**Comment by SeanFeldman** [comment]
@jbogard was it backported to Track 1 SDK (Microsoft.Messaging.ServiceBus)?

**Comment by jbogard** [comment]
Ah no it was not, this would require moving to Azure.Messaging.ServiceBus.

I think this can be closed, I don't think that SDK is getting updated (unlike, say Microsoft.Azure.Cosmos that was upgraded)

**Comment by jbogard** [comment]
Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider is another example of a package (not Azure SDK) that was rev'd a major version to switch over to Azure.Identity.

It's safe to assume that Microsoft.Azure.* libraries will NOT be migrated to Azure.Identity. Azure.* libraries use Azure.Identity.

This package would have to swap to Azure.Messaging.ServiceBus, which is a separate issue, then you get Azure.Identity for free.

**Comment by SeanFeldman** [comment]
I've added #361.
You can mention support for Azure.Identity there as an additional reason to support the new SDK.

**Comment by SeanFeldman** [comment]
Requires #361

**Comment by mderriey** [comment]
In the meantime, we've leveraging the token provider feature of Microsoft.Azure.ServiceBus to use Azure.Identity to achieve AAD authentication to Azure Service Bus.

More details over at <https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/issues/361#issuecomment-932860191>.
